### ISSUE TITLE
Addon exits without an error message 

### ISSUE THREAD

**Issue by staticlama** [pergolafabio/Hikvision-Addons#38]
Hi there, 

Thank you for the work on this addon. I've installed the addon in homeassistant but it does not stay running (similar to the issue raised here in issue _#24 - sdk container exits without alert_). The sensor states have never updated so it looks like it's failing before querying the device.

The logs show:
`Using OS: Linux with architecture: x86_64
02/06/2023, 06:56:27 Hikvision SDK Add-on started! Listening for events...
loop[2] find 10 mac and 16 ip`

Any assistance would be appreciated!

Thank you

**Comment by pergolafabio** [comment]
Hmm, try rebooting your Hikvision device? First time it reads a backlog, so it can be stuck

**Comment by pergolafabio** [comment]
hey, did you figure out? i have another user complaining about same issue, addon stops directly

**Comment by staticlama** [comment]
Not yet. I tried rebuilding the addon and restarting the device. It’s definitely not a user and/password or connectivity issue as I do get a proper error message if those details are incorrect. 

I don’t have an indoor station at the moment so I’m not sure if that’s contributing to the issue. I’ve set the indoor station ip to the same as the intercom in the addon config

**Comment by pergolafabio** [comment]
Yes, someone else had the same problem, addon just stops without an error... Indeed if you fill on purpose a wrong password or IP a decent error shows up...
As a test I helped him, he opened port 8000 and gave me his public IP....

I connect to his doorbell on public IP and add stays started...
But if he tries local, same issue..
Very strange, not sure what's happening

**Comment by pergolafabio** [comment]
The indoor doesn't mather , that IP is optional, only used for sending callsignal with the stdin service

**Comment by staticlama** [comment]
That’s a weird one to debug! Are you able to increase the level of logging in the addon? I have a separate raspberry pi running HomeAssistant - I’ll try install the addon there. My main instance I’ve been using so far is running on a dedicated hp thinclient.

**Comment by pergolafabio** [comment]
Not sure where to debutg it looks it doesn't even try to login... 
I think his outdoor station have locked the IP of the HA instance, maybe if he waits long enough, it will work eventually... That maybe explains why I was able to connect with my remote IP and he was not able to login with his local IP...

Buts a guess..

Maybe try indeed the RPI approach..

Gonna see if I can add some debug

**Comment by pergolafabio** [comment]
I maybe have an idea what's wrong, I'm using docker python 3.10-slim image... Maybe newer installs like you, they pull latest docker image with maybe updated python version, maybe something changed in python that makes the docker crash and that explains why my addon still works...

Can you test something? Can you maybe fork my addon? 

Then command out line 130 and 131 in hik.py and uncomment 135 136, it's a different way of logging in ..

I know some python versions did a segment fault...

Then add your GitHub as a custom repository and install it

**Comment by staticlama** [comment]
No problem - I've created a custom repository with the changes you suggested but I'm still having the same issue. Can't test on raspberry pi as mine is a Pi3 (no x64). Do I need to have anything enabled on the door station itself?

**Comment by pergolafabio** [comment]
I have a separate addon for RPI, it's based on aarch64...

But I check tomorrow , gonna see what python version is included in my addon and then revert back to that specific version (if that's the issue)

**Comment by pergolafabio** [comment]
No, nothing should be enabled ob door intercom, it logs in the same was as ivms does, just admin credentials are needed

**Comment by pergolafabio** [comment]
Maybe a quick test, can you uninstall addon, and change dockerfile, the image to :

python:3.10.5-slim

I believe now it's just : python:3.10-slim

Revert the previous changes also

**Comment by staticlama** [comment]
Thanks for your help! Looks like we're making progress, made the changes, and I'm now receiving a new error prior to the container quitting.

`Using OS: Linux with architecture: x86_64
02/07/2023, 07:58:25 Hikvision SDK Add-on started! Listening for events...
loop[2] find 10 mac and 16 ip
NET_DVR_SetupAlarmChan_V41 failed, error code = 1924`

**Comment by pergolafabio** [comment]
Ok, indeed some progress, gonna check tomorrow my python version of addon

**Comment by pergolafabio** [comment]
Do you had watchdog enabled on addon? Maybe it was constantly restarting and hammering your Hikvision device when addon was crashing
I think that error means : max_exceed ...

Can you try rebooting the Hikvision device?

**Comment by pergolafabio** [comment]
It's also possible since it maybe works , it's downloading a huge backlog from your device now , I know it was sometimes an issue with other users, and rebooting device and or restarting addon helps untill backlog is complete gone from device

**Comment by staticlama** [comment]
I did have watchdog enabled briefly yesterday so that may be it. I’ve restarted the device and addon and have the same error but maybe it needs time to reset the rate limit. I’ll try again later today

**Comment by pergolafabio** [comment]
Ok I'll do some test also tomorrow, Sleeptime now, 23:20 here :-)

**Comment by staticlama** [comment]
Thanks for your help! I haven't made any further changes but am receiving a new error (and the container continues to run!) so definitely seems like things are working better now. Although sensor states are not yet updating (i.e. no changes at all):

`Using OS: Linux with architecture: x86_64
02/07/2023, 09:53:13 Hikvision SDK Add-on started! Listening for events...
loop[2] find 10 mac and 16 ip
02/07/2023, 09:53:14 Unhandled command: 20482
02/07/2023, 09:53:14 Unhandled command: 20482
02/07/2023, 09:53:14 Unhandled command: 20482
02/07/2023, 09:53:14 Unhandled command: 20482
02/07/2023, 09:53:14 Unhandled command: 20482
02/07/2023, 09:53:14 Unhandled command: 20482
02/07/2023, 09:53:25 Unhandled command: 20482
02/07/2023, 09:53:30 Unhandled command: 20482
02/07/2023, 09:55:28 Unhandled command: 20482
02/07/2023, 10:00:34 Unhandled command: 20482
02/07/2023, 10:05:40 Unhandled command: 20482
02/07/2023, 10:05:45 Unhandled command: 20482
02/07/2023, 10:05:50 Unhandled command: 20482
02/07/2023, 10:09:19 Unhandled command: 20482
02/07/2023, 10:09:19 Unhandled command: 20482
02/07/2023, 10:09:26 Unhandled command: 20482
02/07/2023, 10:09:35 Unhandled command: 20482
02/07/2023, 10:09:40 Unhandled command: 20482
02/07/2023, 10:09:54 Unhandled command: 20482
02/07/2023, 10:09:54 Unhandled command: 20482
02/07/2023, 10:10:01 Unhandled command: 20482
02/07/2023, 10:10:10 Unhandled command: 20482
02/07/2023, 10:10:15 Unhandled command: 20482
02/07/2023, 10:10:45 Unhandled command: 20482
02/07/2023, 10:14:17 Unhandled command: 20482
02/07/2023, 10:14:40 Unhandled command: 20482
02/07/2023, 10:14:40 Unhandled command: 20482
02/07/2023, 10:14:45 Unhandled command: 20482
02/07/2023, 10:14:50 Unhandled command: 20482`

**Comment by pergolafabio** [comment]
I think that's huge backlog with errors, is it still doing that?

**Comment by pergolafabio** [comment]
Maybe those errors you see are the failed login attemps caused by watchdog restarting the addon constantly

**Comment by pergolafabio** [comment]
Because unhandled command isn't actual an errors , it just means another event that I just dismiss, because it's none of the motion/call/door sensor... 
What type of device do you have?

**Comment by pergolafabio** [comment]
seems my addon was running on python version 3.10.8 
gonna see if i do a reinstall ...

![image](https://user-images.githubusercontent.com/43251136/217174245-ef8679dd-ab94-4646-ab44-12b11313aa4b.png)

**Comment by staticlama** [comment]
I’m back to getting this error with the container quitting. Have restarted both the device and addon again but still have the same issue:
`NET_DVR_SetupAlarmChan_V41 failed, error code = 1924 `

I have a DS-K1T341AM door controller.

**Comment by staticlama** [comment]
Not sure if I’ve hit a rate limit again as I made no other changes :(

**Comment by pergolafabio** [comment]
ah, but maybe that device is not supported? anyway, i know why the addon stops, seems when pulling the latest docker image it uses python 3.10.9 instead of 3.10.8 , and startig the script gives segmentation fault ... the issue i was taling about before

![image](https://user-images.githubusercontent.com/43251136/217182563-907f778d-898d-4137-adc6-a2039f481300.png)

**Comment by staticlama** [comment]
Oh that’s good news you’ve found the issue, is that easy to fix in the container? Yeah I’m not sure if my device is supported but it is compatible with the gen2 intercom range and someone in the very long HomeAssistant community thread for your addon had the same device and from their question it suggested it was at least partially working for them so I am hopeful.

**Comment by pergolafabio** [comment]
:-)

yeah, i can have a look later... but difficult for me since i only have 1 device to test on ...

anyway, reverted image to 3.10.8-slim, and now it stays running

![image](https://user-images.githubusercontent.com/43251136/217185274-bce0d3d9-86f8-4852-b427-af67395ac89c.png)

**Comment by pergolafabio** [comment]
is your device a new device? maybe the sdk lib i'm using is not compatible? i'm still using the older version
As a test you can download the linx64 from here: https://www.hikvision.com/en/support/download/sdk/
extract it , and you see a "lib" folder, just use those files inside instead of the ones i'm using

**Comment by pergolafabio** [comment]
i believe i'm using 6.1.8 or even older

**Comment by pergolafabio** [comment]
gonna close ticket, issue is resolved
for your device, the event 20482 => hex 5002
https://open.hikvision.com/hardware/definitions/NET_DVR_SetDVRMessageCallBack_V30.html  

5002  = NET_DVR_ACS_ALARM_INFO

Is/was is some of the event below maybe?
https://open.hikvision.com/hardware/structures/NET_DVR_ACS_ALARM_INFO.html

**Comment by pergolafabio** [comment]
The 5002 is already part of the addon:
[kk](https://github.com/pergolafabio/Hikvision-Addons/blob/main/hikvision-sdk/hcnetsdk.py#L93)

i just dont do anything with it in the hik.py script
